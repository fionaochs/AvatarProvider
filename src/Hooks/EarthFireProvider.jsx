import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const EarthFireContext = createContext();

export const EarthFireProvider = ({ children }) => {
  const [earthFire, setEarthFire] = useState('earth');
  
  const toggle = ({ target }) => {
    if(target.checked) setEarthFire('fire');
    if(!target.checked) setEarthFire('earth');
  };

  return (
    <EarthFireContext.Provider value={{ earthFire, toggle }}>
      {children}
    </EarthFireContext.Provider>
  );
};

EarthFireProvider.propTypes = {
  children: PropTypes.node
};

export const useEarthFire = () => {
  const { earthFire } = useContext(EarthFireContext);
  return earthFire;
};

export const useToggle = () => {
  const { toggle } = useContext(EarthFireContext);
  return toggle;
};
