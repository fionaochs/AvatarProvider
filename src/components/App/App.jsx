import React from 'react';
import Header from '../Header/Header';
import Display from '../../Containers/CharactersViewer';

export default function App() {
  return (
    <>
      <Header/>
      <Display/>
    </>
  );
}

