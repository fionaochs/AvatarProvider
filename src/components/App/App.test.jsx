import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { EarthFireProvider } from '../../Hooks/EarthFireProvider';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(
      <EarthFireProvider>
        <App />
      </EarthFireProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
