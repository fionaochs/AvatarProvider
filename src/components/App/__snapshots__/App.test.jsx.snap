// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<ContextProvider
  value={
    Object {
      "earthFire": "earth",
      "toggle": [Function],
    }
  }
>
  <App />
</ContextProvider>
`;
